tests/e2e/lib/virt_helpers.go

```diff
@@ -0,0 +1,408 @@
+func (v *VirtOperator) installHco() error {
+	resourceId := schema.GroupVersionResource{
+		Group:    "hco.kubevirt.io",
+		Resource: "hyperconvergeds",
+		Version:  "v1beta1",
+	}
+
+	unstructuredHco := unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "hco.kubevirt.io/v1beta1",
+			"kind":       "HyperConverged",
+			"metadata": map[string]interface{}{
+				"name":      "kubevirt-hyperconverged",
+				"namespace": v.Namespace,
+			},
+			"spec": map[string]interface{}{},
+		},
+	}
+
+	_, err := v.Dynamic.Resource(resourceId).Namespace(v.Namespace).Create(context.Background(), &unstructuredHco, v1.CreateOptions{})
+	if err != nil {
+		log.Printf("Error creating HCO: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+// Helper function to poll status with a timeout
+func waitForCheck(timeout, period time.Duration, checkFunction func() bool, errorMessage string) error {
+	timer := time.After(timeout)
+	for {
+		if check := checkFunction(); check {
+			break
+		}
+		select {
+		case <-timer:
+			return errors.New(errorMessage)
+		default:
+			time.Sleep(period)
+		}
+	}
+	return nil
+}
+
+// Creates target namespace if needed, and waits for it to exist
+func (v *VirtOperator) ensureNamespace(timeout time.Duration) error {
+	if !v.checkNamespace() {
+		if err := v.installNamespace(); err != nil {
+			return err
+		}
+		if err := waitForCheck(timeout, 1*time.Second, v.checkNamespace, "timed out waiting to create namespace "+v.Namespace); err != nil {
+			return err
+		}
+	} else {
+		log.Printf("Namespace %s already present, no action required", v.Namespace)
+	}
+
+	return nil
+}
+
+// Creates operator group if needed, and waits for it to exist
+func (v *VirtOperator) ensureOperatorGroup(timeout time.Duration) error {
+	if !v.checkOperatorGroup() {
+		if err := v.installOperatorGroup(); err != nil {
+			return err
+		}
+		if err := waitForCheck(timeout, 1*time.Second, v.checkOperatorGroup, "timed out waiting to create operator group kubevirt-hyperconverged-group"); err != nil {
+			return err
+		}
+	} else {
+		log.Printf("Operator group already present, no action required")
+	}
+
+	return nil
+}
+
+// Creates the virtualization subscription if needed, and waits for it to exist
+func (v *VirtOperator) ensureSubscription(timeout time.Duration) error {
+	if !v.checkSubscription() {
+		if err := v.installSubscription(); err != nil {
+			return err
+		}
+		if err := waitForCheck(timeout, 1*time.Second, v.checkSubscription, "timed out waiting to create subscription"); err != nil {
+			return err
+		}
+	} else {
+		log.Printf("Subscription already created, no action required")
+	}
+
+	return nil
+}
+
+// Waits for the ClusterServiceVersion to go to ready, triggered by subscription
+func (v *VirtOperator) ensureCsv(timeout time.Duration) error {
+	return waitForCheck(timeout, 5*time.Second, v.checkCsv, "timed out waiting for CSV to become ready")
+}
+
+// Creates HCO instance if needed, and waits for it to go healthy
```

Comment:
I would use full  HyperConverged Operator in the desc as I had to go back to remember what the HCO stands for :)